Babel 7

Hypnosphi opened the pull request:
Probably needs delaying: https://github.com/storybooks/storybook/pull/3746#issuecomment-403150140

DISCLAIMER: this is about moving Storybook build process itself to Babel 7, and making it the default. You should be able to use Babel 7 in your app with current stable version of Storybook, you just need to install [babel-bridge](https://github.com/babel/babel-bridge)

Issue: #3335 

This doesn't yet fix all the issues linked from there but it performs the necessary breaking change so that rest of the work could be done in a minor release

Hypnosphi committed:
Initial move to babel 7

Hypnosphi committed:
Merge remote-tracking branch 'origin/master' into babel-7

Hypnosphi committed:
Move lib/core to babel@7

Hypnosphi committed:
Move back to .babelrc

Hypnosphi committed:
Fix lint error

vicapow commented:
Woo!! Can't wait for this to land. Anything we can do to help?

Hypnosphi commented:
Thanks @vicapow, I hope to be able to finish this in upcoming days

Hypnosphi committed:
Merge master

Hypnosphi committed:
Update dependencies

augbog commented:
Thanks for all your hard work on this -- looking forward to this :)

ndelangen commented:
@vicapow When this lands in the first alpha, we're going to need a brave soul to test it 😎

sakulstra commented:
@ndelangen we're definitely going to :)
storybook is the last tool in our chain hopping on the babel 7 beta train :D so this will have quite a positive affect on our package.lock

vicapow commented:
Happy to volunteer!

Hypnosphi committed:
Merge master

Hypnosphi commented:
We have to use `babel-preset-minify@canary` for now, see https://github.com/babel/minify/issues/850

Hypnosphi committed:
Add @babel/runtime dependency to all the packages with server-side part

Hypnosphi committed:
Update dependencies

Hypnosphi commented:
@vicapow @augbog I've added a disclamer on top

Hypnosphi committed:
Migration guide

Hypnosphi committed:
Update CLI

Hypnosphi committed:
Fix vue build

Hypnosphi commented:
Things left to do here:

* [ ] RN
* [ ] Jest

Hypnosphi committed:
Update lockfile

Hypnosphi commented:
Looks like https://github.com/facebook/jest/issues/6573 is a blocker here

Hypnosphi committed:
Fix jest

Hypnosphi committed:
Fix jest

shilman commented on lib/cli/test/snapshots/vue/package.json:
```diff
@@ -38,6 +38,6 @@
     "@storybook/addon-actions": "^4.0.0-alpha.12",
     "@storybook/addon-links": "^4.0.0-alpha.12",
     "@storybook/addons": "^4.0.0-alpha.12",
-    "babel-preset-vue": "^2.0.2"
```
@Hypnosphi this looks suspicious?

Hypnosphi replied:
I add it inside app, to be consistent with `@storybook/react`:

https://github.com/storybooks/storybook/pull/3746/files#diff-48d46ebd591f6220a5142b66a694395b

shilman commented on lib/components/src/addon_panel/__snapshots__/index.stories.storyshot:
```diff
@@ -5,29 +5,29 @@ exports[`Storyshots Components|AddonPanel default 1`] = `
     </div>
     <div
-      class="css-18lpvrh"
+      class="css-18lpvrh e1lzzkxx3"
```
@Hypnosphi what's up with all these extra CSS classes? Where do they come from? 🤔

Hypnosphi replied:
No idea, something from emotion probably

Hypnosphi replied:
This PR updates `babel-plugin-emotion` among other things, so that may be it

Hypnosphi commented:
Looks like we can't migrate RN right now. It only supports Babel 7 since yesterday, and tools like CRNA and Expo haven't adapted yet. Let's do this in a separate PR

Hypnosphi committed:
Fix RN test

Hypnosphi committed:
Revert CLI changes for RN

Hypnosphi commented:
I think I'll add a peer dependency on `babel-loader` for easier backwards compatibility

Hypnosphi committed:
Add babel-loader peer dep

Hypnosphi committed:
Commit snapshot

Hypnosphi committed:
Fix version choice logic

Hypnosphi commented:
Babel deprecates stage presets, so we need to pick the particular plugins we want to add, see https://github.com/babel/babel/releases/tag/v7.0.0-beta.52

Looking at the number of breaking changes in each beta, I tend to think that we need to hold off this PR until stable Babel 7 is released. There are no estimations for that, so Storybook v4 should probably be released without this PR

CC @shilman @ndelangen

NameFILIP commented:
😬😬😬

kirill-konshin commented:
Just make sure Storybook can work without quirks on a project based on Babel 7. I suggest to somehow make it easily pluggable. Currently we can reconfigure Webpack, but that's quite awkward.

Hypnosphi commented:
Storybook should work as is with babel-bridge, given that you have a `.babelrc` file

sneko commented:
@Hypnosphi before I had this error:
```
info @storybook/vue v3.4.8
info
=> Loading custom .babelrc
=> Loading custom addons config.
=> Using default webpack setup based on "vue-cli".
info Building storybook ...
ERR! Failed to build the storybook
ERR! ./.storybook/config.js
ERR! Module build failed: TypeError: Cannot read property 'bindings' of null
ERR!     at Scope.moveBindingTo (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\scope\index.js:978:13)
ERR!     at BlockScoping.updateScopeInfo (E:\XXXXXXXXXXXXXXXXXX\node_modules\babel-plugin-transform-es2015-block-scoping\lib\index.js:364:17)
ERR!     at BlockScoping.run (E:\XXXXXXXXXXXXXXXXXX\node_modules\babel-plugin-transform-es2015-block-scoping\lib\index.js:330:12)
ERR!     at PluginPass.BlockStatementSwitchStatementProgram (E:\XXXXXXXXXXXXXXXXXX\node_modules\babel-plugin-transform-es2015-block-scoping\lib\index.js:70:24)
ERR!     at newFn (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\visitors.js:237:21)
ERR!     at NodePath._call (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\path\context.js:65:20)
ERR!     at NodePath.call (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\path\context.js:40:17)
ERR!     at NodePath.visit (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\path\context.js:100:12)
ERR!     at TraversalContext.visitQueue (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\context.js:142:16)
ERR!     at TraversalContext.visitSingle (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\context.js:102:19)
ERR!     at TraversalContext.visit (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\context.js:182:19)
ERR!     at Function.traverse.node (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\index.js:106:17)
ERR!     at traverse (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\traverse\lib\index.js:76:12)
ERR!     at transformFile (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\core\lib\transformation\index.js:116:29)
ERR!     at runSync (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\core\lib\transformation\index.js:45:3)
ERR!     at transformSync (E:\XXXXXXXXXXXXXXXXXX\node_modules\@babel\core\lib\transform-sync.js:17:38)
ERR!  @ multi ./node_modules/@storybook/vue/dist/server/config/polyfills.js ./node_modules/@storybook/vue/dist/server/config/globals.js ./.storybook/config.js
```

After looking on forums I understand that there was an issue of babel versions. Now I'm trying to set the "bridge" by following their README.md:
https://github.com/babel/babel-bridge

But I get know this kind of error:
```
info @storybook/vue v3.4.8
info
=> Loading custom .babelrc
=> Loading custom addons config.
=> Using default webpack setup based on "vue-cli".
info Building storybook ...
ERR! Failed to build the storybook
ERR! ./node_modules/@storybook/vue/dist/client/preview/NoPreview.vue?vue&type=style&index=0&lang=css
ERR! Module parse failed: Unexpected token (21:0)
ERR! You may need an appropriate loader to handle this file type.
ERR! |
ERR! |
ERR! | .nopreview_wrapper {
ERR! |   position: fixed;
ERR! |   display: flex;
ERR!  @ ./node_modules/@storybook/vue/dist/client/preview/NoPreview.vue 4:0-68
ERR!  @ ./node_modules/@storybook/vue/dist/client/preview/render.js
ERR!  @ ./node_modules/@storybook/vue/dist/client/preview/index.js
ERR!  @ ./node_modules/@storybook/vue/dist/client/index.js
ERR!  @ ./.storybook/config.js
ERR!  @ multi ./node_modules/@storybook/vue/dist/server/config/polyfills.js ./node_modules/@storybook/vue/dist/server/config/globals.js ./.storybook/config.js
```

Is there potentially something I forgot to do?

Thanks

ndelangen committed:
Merge branch 'master' into babel-7

# Conflicts:
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/events/package.json
#	addons/info/package.json
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/options/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/storyshots/storyshots-puppeteer/package.json
#	addons/storysource/package.json
#	addons/viewport/package.json
#	app/angular/package.json
#	app/html/package.json
#	app/marko/package.json
#	app/mithril/package.json
#	app/polymer/package.json
#	app/react/package.json
#	app/vue/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/html-kitchen-sink/package.json
#	examples/marko-cli/package.json
#	examples/mithril-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/vue-kitchen-sink/package.json
#	lib/cli/package.json
#	lib/cli/test/snapshots/angular-cli/package.json
#	lib/cli/test/snapshots/marko/package.json
#	lib/cli/test/snapshots/meteor/package.json
#	lib/cli/test/snapshots/mithril/package.json
#	lib/cli/test/snapshots/polymer/package.json
#	lib/cli/test/snapshots/react/package.json
#	lib/cli/test/snapshots/react_native/package.json
#	lib/cli/test/snapshots/react_native_scripts/package.json
#	lib/cli/test/snapshots/react_project/package.json
#	lib/cli/test/snapshots/react_scripts/package.json
#	lib/cli/test/snapshots/sfc_vue/package.json
#	lib/cli/test/snapshots/update_package_organisations/package.json
#	lib/cli/test/snapshots/vue/package.json
#	lib/cli/test/snapshots/webpack_react/package.json
#	lib/core/package.json
#	yarn.lock

ndelangen committed:
FIX snapshots

ndelangen commented:
I'm trying to get this back into action, merging in master and testing..

ndelangen committed:
Merge branch 'master' into babel-7

ndelangen committed:
FIX snapshots

ndelangen committed:
REMOVE smoke-tests in cli (static build examples are now all in teamcity)

ndelangen committed:
Merge branch 'master' into babel-7

# Conflicts:
#	yarn.lock

ndelangen commented:
@Hypnosphi I think it's green now. Do you have a list of remaining tasks on this or after this?

pksunkara commented:
Guys, there's some ambiguity regarding this. Can someone confirm whether this will be part of v4 or not?

ndelangen commented:
@pksunkara we're discussing what to do. I got recommended by the babel team to hold off releasing this as latest until they release a release candidate.

We might do duo releases until then, @shilman seemed to be up for doing that. It seems to be quite in demand 👍.

But obviously the reason we're careful here is because this could break a lot of people's setup and make it hard for them to upgrade. Considering the babel team has (AFAIK) 2 more breaking changes planned in the beta, we don't want to burden our users with a lot of migrations.

@Hypnosphi proposed to wait until create-react-app migrates too.

pksunkara commented:
> We might do duo releases until then

Something like `storybook@babel7`?

danielduan commented:
I'm gonna start on https://github.com/storybooks/babel-plugin-react-docgen/pull/38 then

ndelangen commented:
> Something like storybook@babel7?
Exactly!

☝️? @shilman @Hypnosphi

Hypnosphi commented:
`^4.0.0-babel7` will match `4.0.0` which is probably not what we want.

And we can't move from 6 to 7 without a breaking change, so it probably should just be `@storybook/react@5.0.0-alpha`

chrisnojima commented:
@ndelangen just FYI: https://github.com/babel/babel/releases/tag/v7.0.0-rc.1

sneko commented:
@ndelangen what does it imply? This PR can now be merged with just a few changes or is there some reasons to wait a few more weeks? Thanks

Hypnosphi commented:
At the very least, we should get rid of the stage presets

pksunkara commented:
I think a 4.0.0-beta release is the correct way to go once this is merged.

danielduan commented on .babelrc:
```diff
@@ -1,16 +1,34 @@
-      ["transform-runtime", {
+  "presets": [
+    "@babel/preset-env",
+    ["@babel/preset-stage-0", { "decoratorsLegacy": true }],
```
this no longer exists on RC versions

danielduan commented:
I published `babel-plugin-react-docgen@2.0.0-babel7.0` for the upgrade. Please integrate it if you can and let me know if it doesn't work.

ndelangen committed:
Merge branch 'master' into babel-7

# Conflicts:
#	MIGRATION.md
#	addons/backgrounds/package.json
#	addons/events/package.json
#	addons/info/package.json
#	addons/info/src/__snapshots__/index.test.js.snap
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/options/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/storyshots/storyshots-puppeteer/package.json
#	addons/storysource/package.json
#	addons/viewport/package.json
#	app/angular/package.json
#	app/html/package.json
#	app/marko/package.json
#	app/mithril/package.json
#	app/polymer/package.json
#	app/react/package.json
#	app/vue/package.json
#	examples/angular-cli/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/html-kitchen-sink/package.json
#	examples/marko-cli/package.json
#	examples/mithril-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/official-storybook/stories/__snapshots__/addon-info.stories.storyshot
#	examples/official-storybook/stories/__snapshots__/other-dirname.stories.storyshot
#	examples/polymer-cli/package.json
#	examples/vue-kitchen-sink/package.json
#	jest.config.js
#	lib/cli/package.json
#	lib/cli/test/snapshots/angular-cli/package.json
#	lib/cli/test/snapshots/marko/package.json
#	lib/cli/test/snapshots/meteor/package.json
#	lib/cli/test/snapshots/mithril/package.json
#	lib/cli/test/snapshots/polymer/package.json
#	lib/cli/test/snapshots/react/package.json
#	lib/cli/test/snapshots/react_project/package.json
#	lib/cli/test/snapshots/react_scripts/package.json
#	lib/cli/test/snapshots/sfc_vue/package.json
#	lib/cli/test/snapshots/update_package_organisations/package.json
#	lib/cli/test/snapshots/vue/package.json
#	lib/cli/test/snapshots/webpack_react/package.json
#	lib/codemod/package.json
#	lib/components/src/layout/__snapshots__/index.stories.storyshot
#	lib/core/package.json
#	lib/core/src/server/config/webpack.config.prod.js
#	package.json
#	yarn.lock

ndelangen committed:
UPDATE "babel-plugin-react-docgen": "^2.0.0-babel7.0",

ndelangen committed:
REVERT incorrect removal of package relating to angular

ndelangen committed:
REMOVE unneeded tslint config

ndelangen committed:
FIX snapshots

ndelangen committed:
UPGRADE to babel7-rc.1

ndelangen committed:
REMOVE stage presets and replace with plugins we need

ndelangen committed:
FIX eslint

ndelangen committed:
FIX cli-tests

ndelangen commented:
I've upgraded to latest rc, fixed all deprecations

I had to regenerate the yarn lockfile as it become 'corrupt'
which in turn meant a lot of little tweaks necessary apparently

I haven't done runtime checks yet, or tested what effect his has on browser support

ndelangen committed:
FIX cli test snapshots

ndelangen committed:
FIX angular example with a hack?

ndelangen committed:
REVERT changes to angular require-context (BROKEN)

ndelangen committed:
FIX a warning in polymer storybook

igor-dv committed:
Fix jest-ts-babel.js

igor-dv committed:
Try to fix CLI

ndelangen committed:
CHANGE tot use babel7 everywhere (not just for our own build, but also runtime in apps

ndelangen committed:
SORT package.json files

ndelangen committed:
FIX cli snapshots

Hypnosphi commented on addons/actions/package.json:
```diff
@@ -20,10 +20,10 @@
   },
   "dependencies": {
+    "@babel/runtime": "^7.0.0-rc.1",
     "@storybook/addons": "4.0.0-alpha.16",
```
Not actually needed here, it's in resolve.alias

ndelangen committed:
FIX incorrect dependency in app/html

ndelangen committed:
FIX lockfile

ndelangen committed:
TEMP FIX for hyperapp in netlify

Hypnosphi commented on .eslintrc.js:
```diff
@@ -79,6 +79,25 @@ module.exports = {
+        ignoreRefs: true,
+        allowArrowFunctions: true,
+        allowFunctions: true,
+        allowBind: true,
```
Is there anything left that it still disallows?

BTW this rule is about performance so we probably shouldn't shut it up like this

Hypnosphi commented on MIGRATION.md:
````diff
@@ -72,6 +74,20 @@ The `@storybook/react-native` had built-in addons (`addon-actions` and `addon-li
+
+  * If you aren't using Babel yourself, and don't have .babelrc, install following dependencies:
+    ```
+    npm i -D babel-core@bridge @babel/core babel-loader@next
````
now it's only babel-loader

ndelangen replied:
done

Hypnosphi commented on addons/info/src/components/types/PrettyPropType.js:
```diff
@@ -1,8 +1,6 @@
 import React from 'react';
 
-// TODO: FIX this dependency cycle
-/* eslint-disable import/no-cycle */
```
ESLint packages update seems quite irrelevant to this PR, and it makes it harder to review =(

Hypnosphi commented on app/html/package.json:
```diff
@@ -21,11 +21,17 @@
     "html-loader": "^0.5.5",
-    "react": "^16.4.2",
-    "react-dom": "^16.4.2"
+    "react": "^16.4.0",
```
Why downgrade to less secure version?

Hypnosphi commented on app/react/src/server/wrapDefaultBabelConfig.js:
```diff
@@ -1,4 +1,8 @@
+  presets: [
+    ...config.presets,
+    require.resolve('@babel/preset-react'),
+    require.resolve('@babel/preset-flow'),
```
Actualy we can just use [`babel-preset-react-app`](https://github.com/facebook/create-react-app/tree/next/packages/babel-preset-react-app) here. @norbert WDYT?

pksunkara replied:
I agree with @Hypnosphi

ndelangen committed:
REVERT deletion of smoke tests on cli output && REVERT babel6 cli fixture

ndelangen committed:
REVERT version-change for react in app/html

ndelangen committed:
UPDATE react & react-dom for app/svelte

ndelangen committed:
CHANGE the installBabel to only install babel-loader & RENAME to installBabelLoader

ndelangen committed:
FIX snapshots

ndelangen committed:
FIX lockfile

ndelangen committed:
CHANGE docs for migrating to babel 7

ndelangen committed:
REMOVE unneeded dependency

ndelangen committed:
CHANGE cli tests to e2e

ndelangen committed:
FIX package.json

ndelangen committed:
no message

ndelangen committed:
FIX package.json in root version

ndelangen committed:
FIX verdaccio.yaml

ndelangen committed:
FIX cli/run_tests to install for each run individually

ndelangen committed:
FIX so tests are run in correct folder

ndelangen committed:
FIX so legacy packages *are* proxied

ndelangen committed:
FIX webpack-react cli example

ndelangen committed:
CHANGE meteor generator babel file

I'm not 100% why, I've never seen this 'module:' in babelrc before
Might be legacy or meteor specific?

Seems broken here:
https://teamcity.jetbrains.com/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=1567378&_focus=822

But this did not come up in CircleCI 🤔

Hypnosphi commented on lib/cli/generators/METEOR/index.js:
```diff
@@ -44,14 +44,14 @@ export default async npmOptions => {
     babelrc = JSON5.parse(babelrcContent);
     babelrc.plugins = babelrc.plugins || [];
 
-    if (babelrc.plugins.indexOf('module:babel-root-slash-import') < 0) {
```
I had a warning from Babel 7 that told me to add this `module:`. Is there a chance that now it runs with Babel 6 instead?

Hypnosphi replied:
For some reason, there's no `babel-loader` here: https://github.com/storybooks/storybook/blob/f8332e744b56fd4182eeba86538223f8bf43f139/lib/cli/test/snapshots/meteor/package.json

Hypnosphi replied:
Looks like I just forgot to change METEOR on https://github.com/storybooks/storybook/commit/3c0e3832e169e5e874c1bc774b7b68e652c1c803

Hypnosphi replied:
> But this did not come up in CircleCI

This is because you changed only circleCI config when introducing e2e

ndelangen replied:
Seems to be a completely unnecessary plugin that's 3 years old, and unmaintained..
If everything is green with this gone, I'm just removing it.

Hypnosphi replied:
Please also add `installBabelLoader` in meteor

ndelangen committed:
REMOVE the babel-root-slash-import module completely. It's not used by storybook.

shilman commented:
There's a lot of people clamoring for this change, so here's the 4.0 release plan, based on community sentiment (cross-posted from the `#maintenance` channel on our [slack](https://now-examples-slackin-rrirkqohko.now.sh/))

- After the babel 7 PR is merged, we’ll shift into the release process with one more alpha and then some RC’s to get everything stable
- We might squeeze in a few more small improvements, but webpack4, babel7, and cleanup of 3.x deprecations are the big breaking changes for 4.0
- I’ll be coordinating the release posts and related documentation with everybody starting this weekend

Would love your feedback and suggestions on this!

pksunkara commented on addons/storyshots/storyshots-core/package.json:
```diff
@@ -16,8 +16,8 @@
     "storybook": "start-storybook -p 6006"
   },
   "dependencies": {
+    "@babel/runtime": "^7.0.0-rc.1",
```
I don't think we need this.

pksunkara commented on addons/storyshots/storyshots-puppeteer/package.json:
```diff
@@ -13,8 +13,8 @@
     "prepare": "node ../../../scripts/prepare.js"
   },
   "dependencies": {
+    "@babel/runtime": "^7.0.0-rc.1",
```
I don't think we need this.

pksunkara commented on addons/storysource/package.json:
```diff
@@ -20,9 +20,9 @@
     "prepare": "node ../../scripts/prepare.js"
   },
   "dependencies": {
+    "@babel/runtime": "^7.0.0-rc.1",
```
I don't think we need this.

pksunkara replied:
Didnt realize that, thanks

pksunkara commented on app/html/package.json:
```diff
@@ -21,11 +21,17 @@
+    "@babel/core": "^7.0.0-rc.1",
+    "@babel/runtime": "^7.0.0-rc.1",
     "@storybook/core": "4.0.0-alpha.16",
+    "babel-core": "^7.0.0-bridge.0",
```
Do we really need babel in html app? I am not actually sure why we added it to all apps.

pksunkara commented on app/vue/package.json:
```diff
@@ -22,18 +22,23 @@
   },
   "dependencies": {
+    "@babel/core": "^7.0.0-rc.1",
+    "@babel/runtime": "^7.0.0-rc.1",
```
I don't think we need this.

pksunkara commented on lib/cli/test/fixtures/meteor/package.json:
```diff
@@ -6,7 +6,7 @@
   },
   "dependencies": {
-    "babel-runtime": "^6.20.0",
+    "@babel/runtime": "^7.0.0-rc.1",
```
I don't think we need this.

pksunkara commented on lib/core/package.json:
```diff
@@ -24,14 +29,9 @@
-    "babel-plugin-transform-regenerator": "^6.26.0",
-    "babel-plugin-transform-runtime": "^6.23.0",
-    "babel-preset-env": "^1.7.0",
+    "babel-core": "^7.0.0-bridge.0",
```
Yup, **we** don't use it. So, we don't need it.

pksunkara reviewed (commented):
Just a few things I think we forgot to change.

Hypnosphi commented:
Just not to forget: https://github.com/storybooks/storybook/pull/3746#discussion_r211027591

ndelangen committed:
Merge branch 'master' into babel-7

ndelangen committed:
CHANGE teamcity config to test cli e2e via new script

Hypnosphi commented on .teamcity/OpenSourceProjects_Storybook/buildTypes/OpenSourceProjects_Storybook_CliTest.kt:
```diff
@@ -27,7 +27,7 @@ object OpenSourceProjects_Storybook_CliTest : BuildType({
         script {
             name = "Test"
-            scriptContent = "yarn test --cli"
+            scriptContent = "./scripts/run-e2e.sh"
```
Looks like `yarn test --cli` won't work without local registry anyway, so maybe it should start it always?

ndelangen replied:
yeah that's a fair point.

ndelangen committed:
UPGRADE to rc.2

ndelangen committed:
REVERT ci tests for ci via yarn test --cli

ndelangen committed:
CHANGE to silent installs during e2e

ndelangen committed:
CHANGE to fixed version, otherwise they keep changing when testing

pksunkara committed:
Removed the added @babel/core everywhere

pksunkara committed:
Revert "CHANGE the installBabel to only install babel-loader & RENAME to installBabelLoader"

This reverts commit aadd8f0e0baa90d40c523cc43aaa9a30ada788d8.

pksunkara committed:
Remove unnecessary babel bridge installation in installBabel

pksunkara committed:
Use installBabel for meteor cli

igor-dv commented:
Don't we need to add `@babel/core` to every example now?

Hypnosphi commented:
> Don't we need to add @babel/core to every example now?

Technically we do (and babel-loader if it's missing)

ndelangen committed:
FIX scripts/test.js

pksunkara committed:
Pass registry url to lerna

pksunkara committed:
Fixed cli test issues

Hypnosphi committed:
Cleanup not needed anymore

pksunkara committed:
Added babel-loader to app/svelte

igor-dv committed:
Use 'babel-loader' without "require.resolve"

igor-dv committed:
Try to bring the CLI back

ndelangen commented:
LGTM @igor-dv

igor-dv commented:
Done - https://github.com/storybooks/storybook/pull/4062

sebinsua commented:
Will you publish a new alpha version containing this?

ndelangen commented:
@shilman shall we release an alpha version today?

shilman commented:
Released: https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.17

chawax commented:
I tried this release to make Storybook work with React Native 0.56.0 but I always have a message about unallowed presets. Is there something to change in our project ? A working example somewhere ?

The error message I have when starting Storybook :

```
ERROR in ./storybook/addons.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
Error: Plugin/Preset files are not allowed to export objects, only functions. In /Users/administrateur/xxx/xxx-app/node_modules/babel-preset-react/lib/index.js
    at createDescriptor (/Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/config-descriptors.js:178:11)
    at items.map (/Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/config-descriptors.js:109:50)
    at Array.map (<anonymous>)
    at createDescriptors (/Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/config-descriptors.js:109:29)
    at createPresetDescriptors (/Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/config-descriptors.js:101:10)
    at passPerPreset (/Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/config-descriptors.js:58:96)
    at cachedFunction (/Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/caching.js:33:19)
    at presets.presets (/Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/config-descriptors.js:29:30)
    at mergeChainOpts (/Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/config-chain.js:315:26)
    at /Users/administrateur/xxx/xxx-app/node_modules/@babel/core/lib/config/config-chain.js:278:7
 @ multi ./storybook/addons.js ./node_modules/@storybook/react-native/dist/manager/index.js manager[0]
```

Hypnosphi commented:
We didn't migrate React Native part yet

chawax commented:
OK, thanks for your response. Any idea when it will be migrated ?

chrisnojima commented:
fyi RN 0.57 (in rc) supports babel 7

Hypnosphi commented:
0.56 does as well

Hypnosphi commented:
@Gongreg any ideas?

chawax commented:
I just saw migrating to RN 0.56.0 is not a priority for Expo team : https://blog.expo.io/expo-sdk-v29-0-0-is-now-available-f001d77fadf

Absolutely no chance to see Storybook React Native migrating to Babel 7 before CRNA and Expo do it ?
